drivers/adc/ad4080/ad4080.c

```diff
@@ -0,0 +1,882 @@
+ * @param reg_addr - The register address.
+ * @param reg_val - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4080_write(struct ad4080_dev *dev, uint16_t reg_addr, uint8_t reg_val)
+{
+	uint8_t buff[3];
+
+	if (!dev)
+		return -EINVAL;
+
+	buff[0] = no_os_field_get(BYTE_ADDR_H, reg_addr);
+	buff[1] = no_os_field_get(BYTE_ADDR_L, reg_addr);
+	buff[2] = reg_val;
+
+	return no_os_spi_write_and_read(dev->spi_desc, buff, 3);
+}
+
+/**
+ * @brief Read device register.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param reg_val - The data read from the register.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4080_read(struct ad4080_dev *dev, uint16_t reg_addr, uint8_t *reg_val)
+{
+	int32_t ret;
+	uint8_t buff[3];
+
+	if (!dev)
+		return -EINVAL;
+
+	buff[0] = no_os_field_get(BYTE_ADDR_H, reg_addr) | AD4080_SPI_READ;
+	buff[1] = no_os_field_get(BYTE_ADDR_L, reg_addr);
+
+	ret = no_os_spi_write_and_read(dev->spi_desc, buff, 3);
+	if (ret)
+		return ret;
+
+	*reg_val = buff[2];
+
+	return 0;
+}
+
+/**
+ * @brief Update specific register bits.
+ * @param dev - The device structure.
+ * @param reg_addr - The register address.
+ * @param mask - Specific bits mask.
+ * @param reg_val - The data to be written.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4080_update_bits(struct ad4080_dev *dev, uint16_t reg_addr, uint8_t mask,
+		       uint8_t reg_val)
+{
+	int ret;
+	uint8_t data;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad4080_read(dev, reg_addr, &data);
+	if (ret)
+		return ret;
+
+	data &= ~mask;
+	data |= reg_val;
+
+	return ad4080_write(dev, reg_addr, data);
+}
+
+/**
+ * @brief Software reset the device.
+ * @param dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4080_soft_reset(struct ad4080_dev *dev)
+{
+	int ret;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad4080_update_bits(dev, AD4080_REG_INTERFACE_CONFIG_A,
+				 AD4080_SW_RESET_MSK, AD4080_SW_RESET);
+	if (ret)
+		return ret;
+
+	return ad4080_update_bits(dev, AD4080_REG_INTERFACE_CONFIG_A,
+				  AD4080_SW_RESET_MSK, 0);
+}
+
+/**
+ * @brief Set Address Ascension
+ * @param dev - The device structure.
+ * @param addr_asc - The device mode.
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int ad4080_set_addr_asc(struct ad4080_dev *dev, enum ad4080_addr_asc addr_asc)
+{
+	int ret;
+
+	if (!dev)
+		return -EINVAL;
+
+	ret = ad4080_update_bits(dev, AD4080_REG_INTERFACE_CONFIG_A,
+				 AD4080_ADDR_ASC_MSK,
+				 no_os_field_prep(AD4080_ADDR_ASC_MSK, addr_asc));
+	if (ret)
+		return ret;
+
+	dev->addr_asc = addr_asc;
+
+	return 0;
+}
+
+/**
+ * @brief Get Address Ascension
+ * @param dev - The device structure.
+ * @return Address ascension value
+ */
+enum ad4080_addr_asc ad4080_get_addr_asc(struct ad4080_dev *dev)
+{
+	return dev->addr_asc;
```

Comment:
Wouldn't it be safer to read the actual register value, instead of the cached one? This approach seems wrong if any of these fields are written using `ad4080_write`.